infra/scripts/install.sh

```diff
@@ -19,143 +19,187 @@
-                Disables both CUDA version detection and use of --cuda-version.
--h | --help     Show help message and exit
+cat <<EOF
+Usage: ./ccex install [OPTIONS]
+
+--dist                 Install from wheel in ./dist instead of editable mode
+--torch_ver VER        Torch version or family to install.
+                       Accepts:
+                         • 2.5  | 2.6  | 2.7      (family, installs latest)
+                         • 2.6.3, 2.7.0+cu118 ... (exact)
+                         • nightly
+                       Default: ${DEFAULT_FAMILY}
+--cuda_ver MAJ.MIN     Override detected CUDA version (e.g. 12.1)
+--cpu_only             Force CPU-only Torch installation
+                       (disables CUDA detection / --cuda_ver)
+-h | --help            Show this help
 EOF
 }
 
-options=$(getopt -o h --long dist,torch_ver:,cuda_ver:,cpu_only,help -- "$@")
-
-[ $? -eq 0 ] || {
-    echo "Incorrect options provided"
-    exit 1
-}
-
+###############################################################################
+# Option parsing
+###############################################################################
 _DIST=0
-_TORCH_VER="2.6"
-_USER_SPECIFIED_CUDA=""
+_TORCH_VER="${DEFAULT_FAMILY}"
+_USER_CUDA=""
 _CPU_ONLY=false
-_NIGHTLY=""
 
+options=$(getopt -o h --long dist,torch_ver:,cuda_ver:,cpu_only,help -- "$@") || {
+  echo "[ERROR] Invalid command-line options" >&2; exit 1; }
 eval set -- "$options"
 
 while true; do
-    case "$1" in
-        --dist)
-            _DIST=1
-            ;;
-        --torch_ver)
-            _TORCH_VER="$2"
-            shift
-            ;;
-        --cuda_ver)
-            _USER_SPECIFIED_CUDA="$2"
-            shift
-            ;;
-        --cpu_only)
-            _CPU_ONLY=true
-            shift
-            ;;
-        -h|--help)
-            show_help
-            exit 0;
-            ;;
-        --)
-            shift
-            break;;
-        *)  break;;
-    esac
-    shift
+  case "$1" in
+      --dist)        _DIST=1 ;;
+      --torch_ver)   _TORCH_VER="$2"; shift ;;
+      --cuda_ver)    _USER_CUDA="$2"; shift ;;
+      --cpu_only)    _CPU_ONLY=true ;;
+      -h|--help)     show_help; exit 0 ;;
+      --)            shift; break ;;
+      *)             echo "[ERROR] Unknown option $1"; exit 1 ;;
+  esac
+  shift
 done
 
-# Check torch version
-if [ "$_TORCH_VER" != "2.5" ] && [ "$_TORCH_VER" != "2.6" ] && [ "$_TORCH_VER" != "nightly" ]; then
-    echo "Invalid torch version '$_TORCH_VER'"
-    echo "(Use --help to see available torch versions)"
-    exit 1
+###############################################################################
+# Detect (and maybe keep) any existing torch installation
+###############################################################################
+python - <<'PY' >/tmp/.tico_torch_ver 2>/dev/null
+import sys, importlib.util, re, json, os
+spec = importlib.util.find_spec("torch")
+if spec is None:
+    sys.exit(0)
+import torch  # noqa: E402
```

Comment:
```suggestion
import torch
```

Reply:
I just rewrote the script instead, which prevents from using `jq` command.